docs: update ISD and AS Numbering section (#4922)
scionproto/scion: doc/overview.rst

SCION decouples :term:`endpoint` addressing from inter-domain routing.
Routing is based on the ISD-AS tuple, agnostic of endpoint addressing.

An ISD-AS number is 64 bits, with the top 16 bits indicating the ISD, and the bottom 48 bits indicating the AS. The text representation uses a ``-`` separator between the ISD and AS numbers. For example, ``4-ff00:1:f``.

> ISD and AS numbering is described in Section 1.5 "Addressing" of the `Control Plane Internet Draft <https://scionassociation.github.io/scion-cp_I-D/draft-dekater-scion-controlplane.html>`_. SCION ISD and AS numbers are assigned and registered by the `SCION Registry <https://www.scion.org/registry>`_, operated by the `SCION Association <https://www.scion.org>`_.

Reviewer: ```suggestion
ISD and AS numbering is described in Section 1.5 "Addressing" of the `Control Plane Internet Draft <https://www.ietf.org/archive/id/draft-dekater-scion-controlplane-18.html#name-addressing>`_. SCION ISD and AS numbers are assigned and registered by the `SCION Registry <https://www.scion.org/registry>`_, operated by the `SCION Association <https://www.scion.org>`_.
```

ISD Numbers
"""""""""""

ISD numbers are 16-bit identifiers, formatted as decimal.

.. list-table:: ISD Allocations
   :header-rows: 1
   :widths: 20 80

   * - ISD
     - Description
   * - 0
     - The wildcard ISD.
   * - 1 - 15
     - Reserved for documentation and sample code (analogous to
       `RFC 5398 <http://www.iana.org/go/rfc5398>`_).
   * - 16 - 63
     - Private use (analogous to `RFC 6996 <http://www.iana.org/go/rfc6996>`_) -- can be used
       for testing and private deployments.
   * - 64 - 4094
     - Public ISDs. They must be globally unique.

>    * - 4095
>      - Reserved.
>    * - 4096 - 65535

Reviewer: ```suggestion
   * - 4095 - 65535
```
4095  was merged to the following unallocated block.  discussed and unified during IETF draft editing

In order to provide easy comparison with BGP AS numbers, any AS number in the BGP range (0 to 2\ :sup:`32` - 1) is represented as *decimal*. While it is legal to write a BGP AS number using the SCION AS syntax, programs should always use the decimal representation for display. For example, if a program receives ``0:1:f`` it would display it as ``65551``.

.. list-table:: AS Allocations
   :header-rows: 1
   :widths: 30 15 55

   * - AS
     - Size
     - Description
   * - 0
     - 1
     - The wildcard AS.
   * - 1 - 4294967295 (~\ ``0:0:0/16``)
     - ~4.3 billion
     - Mapped to 32-bit `BGP AS numbers <https://www.iana.org/assignments/as-numbers/>`_,
       formatted as decimal. A BGP AS deploying SCION should use the same AS number for both
       BGP and SCION.
   * - ``1:0:0`` - ``1:ffff:ffff``
     - ~4.3 billion
     - Unallocated.
   * - ``2:0:0`` - ``2:ffff:ffff``
     - ~4.3 billion
     - Public SCION AS numbers -- ASes created for SCION that are not existing BGP ASes.
   * - ``3:0:0`` - ``feff:ffff:ffff``
     - ~280 trillion
     - Unallocated.
   * - ``ff00:0:0/32``

>      - 65535

Reviewer: ```suggestion
     - 65536
```
